### PR TITLE
[RLlib] Some minor cleanups.

### PR DESCRIPTION
**Pull request by sven1977**
Some minor cleanups:
* Rename RolloutWorker.save/restore into `get_state/set_state` to match Policy API AND to emphasize that we don't write to/read from a file, but rather produce a state dict or return one.
* Better typehints
* Docstrings
* var names

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/algorithm.py [hunk 1]
```diff
@@ -2506,15 +2505,15 @@ def __getstate__(self) -> dict:
 
     def __setstate__(self, state: dict):
         if hasattr(self, "workers") and "worker" in state:
-            self.workers.local_worker().restore(state["worker"])
+            self.workers.local_worker().set_state(state["worker"])
```
Comment (sven1977) [inline rllib/algorithms/algorithm.py:2508]: Align with `Policy.get_state/set_state` API.
Soft-deprecate for now. These methods are not really used by users, more by RLlib internally.

Comment (sven1977) [reply rllib/algorithms/algorithm.py:2508]: Also to emphasize that we are NOT reading from or writing to a file, but rather return a state dict (or use one to set some state).

File: rllib/algorithms/algorithm.py [hunk 2]
```diff
@@ -23,14 +26,10 @@
...
 
-import gym
-import numpy as np
-import pkg_resources
-from packaging import version
-
 import ray
 from ray._private.usage.usage_lib import TagKey, record_extra_usage_tag
 from ray.actor import ActorHandle
+import ray.cloudpickle as pickle
```
Comment (ArturNiederfahrenhorst) [inline rllib/algorithms/algorithm.py:32]: From the cloudpickle repo:
> Using cloudpickle for long-term object storage is not supported and strongly discouraged.

Just as a reminder that, in the long run, we should checkpoint with cloudpickle.

**Review by ArturNiederfahrenhorst — APPROVED** [review]
lgtm!

**Pull request merged**